python-backend/app/api/mines/mine/resources/mine.py

```diff
@@ -280,14 +280,14 @@ def post(self, mine_no_or_guid=None):
     @requires_role_mine_create
     def put(self, mine_no_or_guid):
         data = self.parser.parse_args()
-        tenure = data['tenure_number_id']
-        lat = data['latitude']
-        lon = data['longitude']
-        mine_name = data['name']
-        mine_note = data['note']
-        status = data['mine_status']
-        major_mine_ind = data['major_mine_ind']
-        region = data['mine_region']
+        tenure = data.get('tenure_number_id')
+        lat = data.get('latitude')
+        lon = data.get('longitude')
+        mine_name = data.get('name')
+        mine_note = data.get('note')
+        status = data.get('mine_status')
+        major_mine_ind = data.get('major_mine_ind')
+        region = data.get('mine_region')
```

Comment:
I presume this will not explode because all of these values were bring provided, but same thing. these will have to None 'checked where appropriate' but this PUT method needs to some real TLC.